src/main/drivers/serial_uart.c

```diff
@@ -267,6 +267,12 @@ static void uartWrite(serialPort_t *instance, uint8_t ch)
 {
     uartPort_t *uartPort = (uartPort_t *)instance;
 
+    // Check if the TX line is being pulled low by an unpowered peripheral
+    if (uartPort->checkUsartTxOutput && !uartPort->checkUsartTxOutput(uartPort)) {
```

Comment:
This is called for every byte, so may be worth optimizing  - possibly by testing `uart->txPinState == TX_PIN_MONITOR` first

Reply:
This is also used for VCP in which case `uartPort->checkUsartTxOutput` won't be defined and `uart->txPinState` won't be valid.

I think that a better optimisation of the serial code will be to remove the DMA support which isn't used, but has many checks being performed.